### PR TITLE
Add "Buffer" progress bar

### PR DESCRIPTION
**Pull request by yuehuang010**
## Add a "Buffer" progress bar that will report the BlockingCollection queue length.
image
 - The progress bar title is a placeholder, please help suggest a good title.

## Add a "Log Processing Statistics" to report the performance of the each type events.
image

## Add a menu option to clear recently opened history.
image

### CODE REVIEW

**Comment by yuehuang010** [comment]
#684 FYI.

**Review by KirillOsenkov — APPROVED** [review]

File: src/StructuredLogger/BinaryLog.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public static IEnumerable<Record> ReadRecords(byte[] binlogBytes)
 
         public static Build ReadBuild(string filePath) => ReadBuild(filePath, progress: null);
 
-        public static Build ReadBuild(string filePath, Progress progress)
+        public static Build ReadBuild(string filePath, Progress progress, Progress bufferUsage = null)
```
Comment (KirillOsenkov) [inline src/StructuredLogger/BinaryLog.cs:30]: Oof, I hadn't realized that this is breaking public API.

Unfortunately I will have to back it out. Sorry about that.

**Comment by KirillOsenkov** [comment]
I merged this PR optimistically, but upon reflection I think I'm going to have to revert it.

1. I'd like to keep the recent items clearing functionality separate from the rest
2. I'd like to keep the buffer capacity tweaks separate from the rest
3. we can't change the public API as there are too many tools using this library for reading binlogs and we can't require all of them to update just for a progress bar
4. our regular users don't need the buffer utilization progress bar nor do they need the Log Processing Statistics
5. I haven't seen a wall clock time improvement for binlog loading, compared with the previous version. So not sure increasing the buffer is observably helping (at least in the windows case)

File: src/StructuredLogger/BinaryLog.cs (hunk 1 above)
Comment (yuehuang010) [reply src/StructuredLogger/BinaryLog.cs:30]: No problem.  Do revert.

**Comment by yuehuang010** [comment]
- I will spin up another PR for the Recent Clear function.
- I haven't found an improvement with buff capacity either, but that could be a Windows or hardware difference.  Is there a way to add a "developer" mode flag?

**Comment by KirillOsenkov** [comment]
Don't worry about it, I just pushed the revert:

https://github.com/KirillOsenkov/MSBuildStructuredLog/commit/efb0a575d68d4da534b5e3ba391cc2d29b86be9f

**Comment by KirillOsenkov** [comment]
I would like to avoid regular users paying any perf costs just for developer monitoring tooling.

Your buffer utilization progress bar is good, as well as the perf counters, but stuff like this is good to keep in the branch perhaps. Feel free to make a branch with the revert of https://github.com/KirillOsenkov/MSBuildStructuredLog/commit/efb0a575d68d4da534b5e3ba391cc2d29b86be9f and use it for local testing.

Don't think we need this in the main branch.

File: src/StructuredLogger/BinaryLog.cs (hunk 1 above)
Comment (paulomorgado) [reply src/StructuredLogger/BinaryLog.cs:30]: Good old overloads are still a thing.

**Pull request merged**